File: backend/src/Asnan.Domain/Entities/ChatMessage.cs
using Asnan.Domain.Common;

namespace Asnan.Domain.Entities;

/// <summary>
/// A single message in a <see cref="ChatConversation"/> — ARCHITECTURE.md
/// §9 (issue #28). Persisted before broadcast ("durable-first") — history
/// is always consistent with what was actually delivered, never the reverse.
/// </summary>
public class ChatMessage : BaseEntity
{
    public Guid ChatConversationId { get; set; }

    public ChatConversation ChatConversation { get; set; } = null!;

    public Guid SenderUserId { get; set; }

    public User SenderUser { get; set; } = null!;

    public string Content { get; set; } = null!;

    public DateTime SentAtUtc { get; set; }
}

File: mobile/lib/features/chat/domain/chat_message.dart
/// Mirrors the backend's `ChatMessageDto`.
class ChatMessage {
  const ChatMessage({
    required this.id,
    required this.chatConversationId,
    required this.senderUserId,
    required this.content,
    required this.sentAtUtc,
  });

  final String id;
  final String chatConversationId;
  final String senderUserId;
  final String content;
  final DateTime sentAtUtc;

  factory ChatMessage.fromJson(Map<String, dynamic> json) => ChatMessage(
        id: json['id'] as String,
        chatConversationId: json['chatConversationId'] as String,
        senderUserId: json['senderUserId'] as String,
        content: json['content'] as String,
        sentAtUtc: DateTime.parse(json['sentAtUtc'] as String),
      );
}
